query/security/wcpm/src/test/groovy/org/codice/ddf/admin/security/wcpm/actions/persist/SaveContextPoliciesTest.groovy

```diff
@@ -0,0 +1,254 @@
+import org.codice.ddf.admin.security.common.fields.wcpm.Realm
+import org.codice.ddf.admin.security.common.services.PolicyManagerServiceProperties
+import org.codice.ddf.admin.security.common.services.StsServiceProperties
+import org.codice.ddf.admin.security.wcpm.actions.WcpmActionCreator
+import org.codice.ddf.security.policy.context.impl.PolicyManager
+import spock.lang.Specification
+
+class SaveContextPoliciesTest extends Specification {
+    ActionCreator actionCreator
+    ConfiguratorFactory configuratorFactory
+    Configurator configurator
+    ConfigReader configReader
+    OperationReport operationReport
+    PolicyManager policyManager
+    Action action
+    Map<String, Object> stsConfig
+
+    String[] testClaims
+    def testData
+
+    def setup() {
+        testClaims = [
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier",
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname",
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname",
+                "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role"
+        ]
+
+        testData = [
+                policies: [
+                        [
+                                paths        : [ '/' ],
+                                authTypes    : [ 'basic' ],
+                                realm        : 'karaf',
+                                claimsMapping: [
+                                        [
+                                                claim     : 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role',
+                                                claimValue: 'system-admin'
+                                        ]
+                                ]
+                        ],
+                        [
+                                paths        : [ '/test' ],
+                                authTypes    : [ 'SAML', 'PKI' ],
+                                realm        : 'karaf',
+                                claimsMapping: [
+                                        [
+                                                claim     : 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role',
+                                                claimValue: 'system-admin'
+                                        ]
+                                ]
+                        ]
+                ]
+        ]
+
+        operationReport = Mock(OperationReport)
+        configuratorFactory = Mock(ConfiguratorFactory)
+        configurator = Mock(Configurator)
+        configReader = Mock(ConfigReader)
+        configurator.updateConfigFile({ it == PolicyManagerServiceProperties.POLICY_MANAGER_PID }, _, _) >> {
+            args -> policyManager.setPolicies(args[1])
+        }
+
+        stsConfig = [(StsServiceProperties.STS_CLAIMS_PROPS_KEY_CLAIMS): testClaims ]
```

Comment:
If anyone is curious, using (<String>) will use the value as a key in a map. It's really handy.